compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaIntersectionTypeSymbol.java

```diff
@@ -73,18 +73,22 @@ public TypeSymbol effectiveTypeDescriptor() {
         TypesFactory typesFactory = TypesFactory.getInstance(this.context);
         BType effectiveType = ((BIntersectionType) this.getBType()).effectiveType;
         this.effectiveType = typesFactory.getTypeDescriptor(effectiveType,
-                                                            effectiveType != null ? effectiveType.tsymbol : null,
-                                                            false, false, true);
+                effectiveType != null ? effectiveType.tsymbol : null,
+                false, false, true);
         return this.effectiveType;
     }
 
     @Override
     public List<FunctionSymbol> langLibMethods() {
         if (this.langLibFunctions == null) {
-            LangLibrary langLibrary = LangLibrary.getInstance(this.context);
-            List<FunctionSymbol> functions = langLibrary.getMethods(
-                    ((AbstractTypeSymbol) this.effectiveTypeDescriptor()).getBType());
-            this.langLibFunctions = filterLangLibMethods(functions, this.getBType());
+            if (this.effectiveTypeDescriptor().typeKind() == TypeDescKind.OBJECT) {
```

Comment:
if `this.effectiveTypeDescriptor().typeKind()` becomes a type reference, it will go to else part wouldn't it?
Can you try with a test
```ballerina
type FooObj object { string target; };

FooObj & readonly readonlyObjIntersec = ...
```

Reply:
Fixed this.